src/sdk/adapter-registry.js

```diff
@@ -0,0 +1,39 @@
+let instance = null
+
+class AdapterRegistry {
+
+  constructor() {
+    if(!instance){
+      instance = this
+      this.registry = {}
+    }
+    return instance
+  }
+
+  valid(adapter) {
+    if(typeof adapter === 'undefined' || !['get', 'all', 'save', 'filter', 'delete']
```

Comment:
A low-hanging-fruit here: you don't need to check if `adapter` is `undefined`. Checking its truthiness is enough and would result on a much cleaner code. Also, you could use a named variable to store the result of the combination of map and reduce:

```js
valid (adapter) {
  var implementAllMethods = ['all', 'delete']
    .map(method => typeof adapter[method] === 'function')
    .reduce((prev, current) => prev && current)

  if (!adapter || !implementAllMethod) {
    throw new Error('Adapter should (...)')
  }
```